### PR TITLE
CB-9557 Fixes apk install failure when switching from debug to release build

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
This fixes [CB-9557](https://issues.apache.org/jira/browse/CB-9557) and probably [CB-8834](https://issues.apache.org/jira/browse/CB-8834)

I've also caught the issue, described on SO (http://stackoverflow.com/questions/26794862/failure-install-failed-update-incompatible) which is resolved by this PR as well.

### CODE REVIEW

File: bin/templates/cordova/lib/device.js [hunk 1]
```diff
@@ -96,10 +96,15 @@ module.exports.install = function(target, buildResults) {
     }).then(function(resolvedTarget) {
         var apk_path = build.findBestApkForArchitecture(buildResults, resolvedTarget.arch);
         var launchName = appinfo.getActivityName();
+        var pkgName = appinfo.getPackageName();
         console.log('Using apk: ' + apk_path);
-        console.log('Installing app on device...');
-        var cmd = 'adb -s ' + resolvedTarget.target + ' install -r "' + apk_path + '"';
-        return exec(cmd, os.tmpdir())
+        console.log('Uninstalling ' + pkgName + ' from device...');
+        return exec('adb -s ' + resolvedTarget.target + ' uninstall ' + pkgName, os.tmpdir())
```
Comment (nikhilkh) [inline bin/templates/cordova/lib/device.js]: Wouldn't uninstalling always cause a failure - especially on the first run?

Comment (vladimir-kotikov) [inline bin/templates/cordova/lib/device.js]: No. Even if `adb uninstall` fails to uninstall an app, it exits with error code 0, and thus `exec` promise is always resolved in this case.

Comment (nikhilkh) [inline bin/templates/cordova/lib/device.js]: Ah - A comment would help clarify that.

Comment (vladimir-kotikov) [inline bin/templates/cordova/lib/device.js]: Done.

**Comment by nikhilkh** [comment]
LGTM

**Pull request merged**